mage_ai/cli/main.py

```diff
@@ -171,55 +179,66 @@ def run(
             sentry_dsn,
             traces_sample_rate=SENTRY_TRACES_SAMPLE_RATE,
         )
-
-    runtime_variables = dict()
-    if runtime_vars is not None:
-        runtime_variables = parse_runtime_variables(runtime_vars)
-
-    sys.path.append(os.path.dirname(project_path))
-    pipeline = Pipeline.get(pipeline_uuid, repo_path=project_path)
-
-    db_connection.start_session()
-
-    if pipeline_run_id is None:
-        default_variables = get_global_variables(pipeline_uuid)
-        global_vars = merge_dict(default_variables, runtime_variables)
-    else:
-        pipeline_run = PipelineRun.query.get(pipeline_run_id)
-        global_vars = pipeline_run.get_variables(extra_variables=runtime_variables)
-
-    if template_runtime_configuration is not None:
-        template_runtime_configuration = json.loads(template_runtime_configuration)
-
-    if block_uuid is None:
-        ExecutorFactory.get_pipeline_executor(
-            pipeline,
-            execution_partition=execution_partition,
-            executor_type=executor_type,
-        ).execute(
-            analyze_outputs=False,
-            global_vars=global_vars,
-            pipeline_run_id=pipeline_run_id,
-            run_sensors=not skip_sensors,
-            run_tests=test,
-            update_status=False,
-        )
-    else:
-        ExecutorFactory.get_block_executor(
-            pipeline,
-            block_uuid,
-            execution_partition=execution_partition,
-            executor_type=executor_type,
-        ).execute(
-            analyze_outputs=False,
-            block_run_id=block_run_id,
-            callback_url=callback_url,
-            global_vars=global_vars,
-            pipeline_run_id=pipeline_run_id,
-            template_runtime_configuration=template_runtime_configuration,
-            update_status=False,
-        )
-    print('Pipeline run completed.')
+    disable_new_relic = False
+    if ENABLE_NEW_RELIC:
+        try:
+            newrelic.agent.initialize(NEW_RELIC_CONFIG_PATH)
+            application = newrelic.agent.register_application(timeout=10)
+        except newrelic.api.exceptions.ConfigurationError as error:
```

Comment:
Is this the only possible exception? Could there be other exceptions?

Reply:
I added a general `except Exception` clause.

Even through there are only two exceptions defined by new relic python API: https://github.com/newrelic/newrelic-python-agent/blob/main/newrelic/api/exceptions.py#L16 It is better to have a general Exception catch here in case new exception introduced by newrelic in future.